### PR TITLE
Add bcftools call.

### PR DESCRIPTION
**Pull request by davebx**

### CODE REVIEW

File: tool_collections/bcftools/bcftools_call/bcftools_call.xml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <expand macro="stdio" />
+    <expand macro="version_command" />
+    <command>
+<![CDATA[
+           #if str($input.ext) == 'vcf':
+
+           cp ${input} input.vcf && bgzip input.vcf &&
```
Comment (bgruening) [inline tool_collections/bcftools/bcftools_call/bcftools_call.xml]: Does `ln` work here as well?

File: tool_collections/bcftools/bcftools_call/bcftools_call.xml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    <description>SNPs and indels</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <expand macro="stdio" />
+    <expand macro="version_command" />
+    <command>
+<![CDATA[
+           #if str($input.ext) == 'vcf':
```
Comment (bgruening) [inline tool_collections/bcftools/bcftools_call/bcftools_call.xml]: proper indentation would be nice :)

**Comment by bgruening** [comment]
@davebx I have a more general question of handling indexed vcf files: What is the proper way to do this? Imho there are more and more tools that are in the need of vcf.gz files and associated indices. Wouldn't it be better to add this to the datatype, similar to BAM?

File: tool_collections/bcftools/bcftools_call/bcftools_call.xml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    </command>
+    <inputs>
+        <expand macro="vcf_input" />
+        <expand macro="filter_conditional" />
+        <expand macro="output_format_selector" />
+        <param label="Caller" name="caller" type="select">
+            <option value="-c">Consensus (-c)</option>
+            <option value="-m">Multiallelic (-m)</option>
+        </param>
+        <param label="Constraint (-C)" name="constraint" type="select">
```
Comment (bgruening) [inline tool_collections/bcftools/bcftools_call/bcftools_call.xml]: brackets should move into help="(-C)"

File: tool_collections/bcftools/bcftools_call/bcftools_call.xml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            <option value="-C alleles">Alleles</option>
+            <option value="-C trio">Trio</option>
+        </param>
+        <param label="Output variants only (-v)" type="boolean" truevalue="-v" falsevalue="" name="variants_only" />
+        <param label="Skip (-V)" name="skip_indels_snps" type="select">
+            <option value="" selected="True">Do not skip</option>
+            <option value="-V indels">Indels</option>
```
Comment (bgruening) [inline tool_collections/bcftools/bcftools_call/bcftools_call.xml]: above you have the the commandline switch at the end in brackets

File: tool_collections/bcftools/bcftools_call/bcftools_call.xml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        </test>
+        <!--test>
+            <param ftype="vcf" name="input_vcf" value="bcftools_call_in2.tab.gz" />
+            <param name="output_format" value="b" />
+            <param name="caller" value="-m" />
+            <output file="bcftools_call_out1.vcf" ftype="vcf" name="output" />
+        </test-->
+    </tests>
+    <help>
+<![CDATA[
```
Comment (bgruening) [inline tool_collections/bcftools/bcftools_call/bcftools_call.xml]: A small sentence what this tool is supposed to do would be nice :)

**Comment by bgruening** [comment]
@davebx do you have time to merge this PR with these tools here: https://github.com/bgruening/galaxytools/tree/master/tools/bcftools/0.2.x

Not sure how updated they are or if useful at all, but if so please feel free to take them over.

**Comment by davebx** [comment]
@bgruening Quite so. I've corrected the ordering.

**Comment by bgruening** [comment]
:+1: anyone else?

**Comment by davebx** [comment]
@bgruening I'm integrating your view wrapper right now, actually. Should be ready to update this as soon as I've worked out a mildly irritating issue with datatypes.

**Comment by bgruening** [comment]
Nice thanks a lot!

**Comment by davebx** [comment]
@bgruening Barring issues with the datatypes I defined and the converters that go with them, I'm ready to sign off on this.

File: tool_collections/bcftools/bcftools_call/bcftools_call.xml [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <expand macro="stdio" />
+    <expand macro="version_command" />
+    <command>
+        <![CDATA[
+            @INPUT_FORMAT@
+            bcftools index ${input} &&
```
Comment (bgruening) [inline tool_collections/bcftools/bcftools_call/bcftools_call.xml]: Input can be one of: format="bcf,bcf_bgzip,vcf,vcf_bgzip"
The indexing step is not required by vcf and for other we can probably use the bcf_index and co.

**Comment by bgruening** [comment]
Thanks Dave for working on this, very appreciated. 
I'm also looking for a `vcf.gz + tbi index` datatype for a new visualisation. This is really useful!

**Comment by erasche** [comment]
Well. Thanks @bgruening for pointing this out to me. xref https://github.com/galaxyproject/tools-iuc/pull/246

File: tool_collections/bcftools/macros.xml [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+                <param label="Output ploidy" name="output_ploidy" type="select">
+                    <option selected="True" value="">Diploid output</option>
+                    <option value="-X">Haploid output for male samples (-X)</option>
+                    <option value="-Y">Haploid output for male samples, skip female (-Y)</option>
+                </param>
+                <param format="ped" label="PED file with regions to include" name="input_regions" type="data" />
+            </when>
+            <when value="targets">
+                <param format="bed,vcf,tabular" label="File listing regions to include" name="input_regions" type="data" />
+            </when>
```
Comment (erasche) [inline tool_collections/bcftools/macros.xml]: Something I noticed from the docs today:

> Both -r and -t options can be applied simultaneously

these aren't actually mutually exclusive options :)

**Comment by erasche** [comment]
Under the provision of this repo that no new tools go into it, @davebx may I merge your work into my open PR in the IUC repo?

**Comment by martenson** [comment]
@erasche :+1:

**Comment by bgruening** [comment]
@davebx @erasche can we close this?
What is the status of the bcf and vcf+index datatypes?

**Comment by erasche** [comment]
I'll work on bcftools wrapping more next week. As for datatypes, no progress has been made.

**Comment by bgruening** [comment]
@davebx was working on it as far as I know.

**Pull request closed without merging**